contracts/handlers/ERC20Handler.sol

```diff
@@ -33,15 +33,26 @@ contract ERC20Handler is IDepositHandler, ERC20Safe {
         _;
     }
 
-    constructor(address bridgeAddress) public {
+    constructor(address bridgeAddress, bytes[] memory initialTokenIDs) public {
         _bridgeAddress = bridgeAddress;
+
+        for (uint256 i = 0; i < initialTokenIDs.length; i++) {
+            address tokenAddress = parseAddressFromTokenID(initialTokenIDs[i]);
+            _tokenIDToTokenContractAddress[initialTokenIDs[i]] = tokenAddress;
+            _tokenContractAddressToTokenID[tokenAddress] = initialTokenIDs[i];
+        }
+    }
+
+    function parseAddressFromTokenID(bytes memory tokenID) internal pure returns (address tokenAddress) {
+        assembly {
+            tokenAddress := mload(add(tokenID, 0x40))
+        }
```

Comment:
This assumes that all token Ids are contract addresses, this is not the case with substrate. We need to be able to specify the contract seperate from the tokenId.

Reply:
I removed the `parseAddressFromTokenID` method and added `initialContractAddresses` argument to [ERC20Handler.sol's constructor](https://github.com/ChainSafe/chainbridge-solidity/blob/9c5a61533adf7d94896c8ea1967d211ab038b113/contracts/handlers/ERC20Handler.sol#L36)